e2e/test/iothub/service/ServiceClientE2ETests.cs

```diff
@@ -34,29 +35,29 @@ public async Task Message_NoTimeoutPassed()
 
         private async Task FastTimeout()
         {
-            TimeSpan? timeout = TimeSpan.FromTicks(10).Negate();
-            await TestTimeout(timeout).ConfigureAwait(false);
+            TimeSpan timeout = TimeSpan.FromTicks(10).Negate();
+            using var cts = new CancellationTokenSource(timeout);
+            await TestTimeout(cts.Token).ConfigureAwait(false);
         }
 
         private async Task DefaultTimeout()
         {
-            TimeSpan? timeout = null;
-            await TestTimeout(timeout).ConfigureAwait(false);
+            await TestTimeout().ConfigureAwait(false);
         }
 
-        private async Task TestTimeout(TimeSpan? timeout)
+        private async Task TestTimeout(CancellationToken token = default)
         {
             using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, DevicePrefix).ConfigureAwait(false);
             using var sender = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
 
             var sw = new Stopwatch();
             sw.Start();
 
-            Logger.Trace($"Testing ServiceClient SendAsync() timeout in ticks={timeout?.Ticks}");
+            Logger.Trace($"Testing ServiceClient SendAsync() cancellation.");
```

Comment:
David has [expressed some interest](https://github.com/Azure/azure-iot-sdk-csharp/pull/2614#discussion_r953150296) in removing these tests

Reply:
There's some value in ensuring that the cancellation token is propagated to the transport layer, but we could probably do that through unit tests as well.

Reply:
Absolutely, should be a unit test to confirm a pre-canceled token aborts the operational call before it is made.